src/SFA.DAS.ApprenticeAccounts/Application/Commands/UpdateAllApprenticePreferencesCommand/UpdateAllApprenticePreferencesCommandHandler.cs

```diff
@@ -0,0 +1,68 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.ApprenticeAccounts.Data;
+using SFA.DAS.ApprenticeAccounts.Data.Models;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeAccounts.Application.Commands.UpdateAllApprenticePreferencesCommand
+{
+    public class UpdateAllApprenticePreferencesCommandHandler : IRequestHandler<UpdateAllApprenticePreferencesCommand>
+    {
+        private readonly IApprenticePreferencesContext _apprenticePreferencesContext;
+        private readonly IApprenticeContext _apprenticeContext;
+        private readonly IPreferencesContext _preferencesContext;
+        private readonly ILogger<UpdateAllApprenticePreferencesCommandHandler> _logger;
+
+        public UpdateAllApprenticePreferencesCommandHandler(IApprenticePreferencesContext apprenticePreferencesContext,
+            IApprenticeContext apprenticeContext, IPreferencesContext preferencesContext, ILogger<UpdateAllApprenticePreferencesCommandHandler> logger)
+        {
+            _apprenticePreferencesContext = apprenticePreferencesContext;
+            _apprenticeContext = apprenticeContext;
+            _preferencesContext = preferencesContext;
+            _logger = logger;
+        }
+
+        public async Task<Unit> Handle(UpdateAllApprenticePreferencesCommand request, CancellationToken cancellationToken)
+        {
+            foreach (var apprenticePreference in request.ApprenticePreferences)
+            {
+                _logger.LogInformation($"Fetch Apprentice by Id. Id used: {apprenticePreference.ApprenticeId}");
+                var apprentice = await _apprenticeContext.Entities.FindAsync(apprenticePreference.ApprenticeId);
+
+                _logger.LogInformation($"Fetch Preference by Id. Id used: {apprenticePreference.PreferenceId}");
+                var preference = await _preferencesContext.Entities.FindAsync(apprenticePreference.PreferenceId);
+
+                if (apprentice == null || preference == null)
```

Comment:
Same as the individual update handler, mediator validation on the command before it hits the handler might not be a bad idea https://medium.com/swlh/validation-with-mediator-pipelines-in-asp-net-core-applications-7878a56ec604